src/renderers/Dialog.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { View, Animated, Easing, StyleSheet } from 'react-native';
+import { OPEN_ANIM_DURATION, CLOSE_ANIM_DURATION } from '../constants';
+
+export default class Dialog extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      fadeAnim: new Animated.Value(0.1),
+    };
+  }
+
+  componentDidMount() {
+    Animated.timing(this.state.fadeAnim, {
+      duration: OPEN_ANIM_DURATION,
+      toValue: 1,
+      easing: Easing.out(Easing.cubic),
+      useNativeDriver: true
+    }).start();
+  }
+
+  close() {
+    return new Promise(resolve => {
+      Animated.timing(this.state.fadeAnim, {
+        duration: CLOSE_ANIM_DURATION,
+        toValue: 0,
+        easing: Easing.in(Easing.cubic),
+        useNativeDriver: true
+      }).start(resolve);
+    });
+  }
+
+  render() {
+    const { style, children, layouts, ...other } = this.props;
+    const animation = {
+      opacity: this.state.fadeAnim,
```

Comment:
what about scale animation as well (similar to context menu)?

Reply:
I tried it and I find it a little bit disturbing. IMHO scaling animation in dialogs is not very common in mobile apps (at least android).